From 3b44e50b4437050f515251cf1a9b8705ddbc4763 Mon Sep 17 00:00:00 2001
From: Jonas Larsson <ljonas@google.com>
Date: Wed, 28 Aug 2019 11:24:01 -0700
Subject: [PATCH 10/17] eglimage: Fix memory leak

Also free the GstEGLImage struct allocated by g_new0.

Fixes #661

Cherry picked from upstream master:
https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/commit/94984e13a9208b0b6d69dea597f42b225ef1f784

Change-Id: I790e5ab89cdf3e7584a5f0a02950e8d0fd237b48
---
 gst-libs/gst/gl/egl/gsteglimage.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gst-libs/gst/gl/egl/gsteglimage.c b/gst-libs/gst/gl/egl/gsteglimage.c
index d020aa67b..498c62371 100644
--- a/gst-libs/gst/gl/egl/gsteglimage.c
+++ b/gst-libs/gst/gl/egl/gsteglimage.c
@@ -143,6 +143,8 @@ _gst_egl_image_free (GstMiniObject * object)
         (GstGLContextThreadFunc) _gst_egl_image_free_thread, image);
     gst_object_unref (image->context);
   }
+
+  g_free (image);
 }
 
 static GstMiniObject *
-- 
2.26.0

